source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -244,10 +244,10 @@ void HeaderToMetadataFilter::writeHeaderToMetadata(Http::HeaderMap& headers,
     absl::optional<std::string> value = rule.selector_->extract(headers);
 
     if (value && proto_rule.has_on_header_present()) {
-      applyKeyValue(std::move(value).value_or(""), rule, proto_rule.on_header_present(),
+      applyKeyValue(value.value_or(""), rule, proto_rule.on_header_present(),
```

Comment:
@rgs1 this isn't unneeded, a `std::move` makes it dispatched to the second signature, so the value in optional will be actually moved:

> `template< class U >
constexpr T value_or( U&& default_value ) &&;`

https://en.cppreference.com/w/cpp/utility/optional/value_or

Reply:
Oh, I see. But is the `std::string&&` in applyKeyValue() is still needed or will copy elision take care of things?

Reply:
re: `std::string&&`, depends on the caller site, `std::string` does accept both value or rvalue reference (which will be moved), so if this is the only call site, it is already a move. having `std::string&&` will enforce that's a move but not a value copy.

Reply:
Gotcha -- let's enforce it then.